### DISCUSSION TITLE
Webgazer and Webgazer-extension

### DISCUSSION THREAD

**Discussion by agustin-penas** [jspsych/jsPsych#3048] [Q&A]
Hi!
I am working on some experiments with webgazer (trying to add blink detection) and I see I am getting less than 30 webgazer outputs per second (around 20-15). My cammera is 30fps so that is what I expected.
Doing some investigation I found you are using `setInterval()` with a `sampling_interval` of 34ms which should be closer to the 30 predictions per second. Is this interval the minimum or like a discrete interval?  (like if it could not run on 34ms then it will run on the 68th ms) 
Also I saw on a previous version you used `state.webgazer.setGazeListener(handleGazeDataUpdate);`, what was the problem? Why did it change?

**Comment by GEJ1** [comment]
@jodeleeuw we are working together with @agustin-penas on this issue. Do you have any idea? 

Can be related with this discussion? https://github.com/jspsych/jsPsych/discussions/1892

Thank you

**Reply by jodeleeuw** [reply]
My recollection is that there is no guarantee that webgazer will process every video frame. If the computation time of generating the gaze prediction (plus anything else that the system is trying to do at that time) takes longer than the interval between frames then I think frames will be dropped.
